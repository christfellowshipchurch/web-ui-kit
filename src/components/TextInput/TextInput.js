import React, { useState } from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faUser, faTimesCircle } from "@fortawesome/pro-regular-svg-icons";
import { faExclamationCircle, faCheckCircle, faBan } from "@fortawesome/pro-light-svg-icons";

  const colors = {
    fail: '#d52158',
    success: '#3ac27f',
    focus: '#c5c5c5',
    disable: 'lightgrey'
  }

  const type = {
    fail: faExclamationCircle,
    success: faCheckCircle,
    focus: faTimesCircle,
    disable: faBan
  }

  const InputIcon = ({ icon, isFocus, isHover, disabled }) => {
     
    let regColor = isFocus || isHover ? "#00AEEF" : "#737374"
    let color = disabled ? colors.disable : regColor

    return(
    <FontAwesomeIcon
        icon={icon}
        size="2x"
        color={color}
        className="p-1"
    />
  )}

const TextInput = ({ success, fail, label, helper, placeholder, disabled = false }) => {

      const [isFocus, setIsFocus] = useState(false);    
      const [isHover, setIsHover] = useState(false);
      const [value, setValue] = useState('')

    let focusColor = isFocus ? '#9aa1a6' : 'rgba(0,0,0,0)'  

    let icon
    let iconColor

    if(success){
        icon = type.success
        iconColor = colors.success
    }
    else if(fail){
        icon = type.fail
        iconColor = colors.fail
    }
    else if(disabled){
        icon = type.disable
        iconColor = colors.disable
    }else{
        icon = type.focus
        iconColor = isFocus ? colors.disable : 'rgba(0,0,0,0)'
    }


    return (
          <div className="container d-flex align-items-center justify-content-center">
            <div className='row'>
              <div className='col'>
              <label 
                className="d-flex text-left pl-4 mb-0"
                style={{color: focusColor}}
              >
                  {label}
                </label>
                <div className="d-flex align-items-center border-bottom">
                  <InputIcon
                    icon={faUser}
                    isFocus={isFocus}
                    isHover={isHover}
                    disabled={disabled}
                  />
                  <input
                    type="text"
                    placeholder={placeholder}
                    onFocus={() => setIsFocus(!isFocus)}
                    onBlur={() => setIsFocus(!isFocus)}
                    onMouseEnter={() => setIsHover(!isHover)}
                    onMouseLeave={() => setIsHover(!isHover)}
                    disabled={disabled}
                    value={value}
                    onChange={(e) => setValue(e.target.value)}
                  />
                  <FontAwesomeIcon
                    icon={ icon }
                    size="2x"
                    color={iconColor}
                    className="p-1"
                    onClick={() => setValue('')}
                  />
                </div>
                <label 
                    className="d-flex text-left pl-4 mb-0"
                    style={{color: focusColor}}
                    >
                  {helper}
                </label>
              </div>
            </div>
          </div>
    );
    }
export default TextInput