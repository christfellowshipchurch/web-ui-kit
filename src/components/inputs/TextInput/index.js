import React, { useState } from 'react'
import PropTypes from 'prop-types'
import {
  faExclamationCircle,
  faCheckCircle,
  faBan,
  faUser,
} from "@fortawesome/pro-light-svg-icons"

import InputContainer from '../inputContainer'
import InputIcon from '../inputIcon'

const TextInput = ({
  label,
  helper,
  placeholder,
  disabled,
  description,
  error,
  success,
  onChange,
  withSuccess,
  value,
  icon,
  ...inputProps
}) => {
  const [focused, setFocused] = useState(false)
  const [hovering, setHovering] = useState(false)
  const [inputValue, setInputValue] = useState(value)
  const prefixColor = (focused || hovering) && !disabled
    ? '#00aeef'
    : disabled ? '#e6e6e6' : '#525252'
  let suffix = withSuccess
    ? { icon: faCheckCircle, color: '#1ec27f' }
    : null

  if (error) {
    suffix = { icon: faExclamationCircle, color: '#cb045b' }
  }

  if (disabled) {
    suffix = { icon: faBan, color: '#e6e6e6' }
  }

  return (
    <InputContainer
      label={label}
      placeholder={placeholder}
      description={error && !disabled ? error : description}
      focused={focused || !!inputValue}
      hasValue={!!inputValue && inputValue !== ''}
      prefix={<InputIcon icon={icon} color={prefixColor} />}
      suffix={suffix && <InputIcon icon={suffix.icon} color={suffix.color} />}
    >
      <input
        type="text"
        defaultValue={value}
        placeholder={placeholder}
        className="w-100 py-0"
        onFocus={() => setFocused(true)}
        onBlur={() => setFocused(false)}
        onMouseEnter={() => setHovering(true)}
        onMouseLeave={() => setHovering(false)}
        onChange={(e) => { setInputValue(e.target.value); onChange(e) }}
        disabled={disabled}
        {...inputProps} />
    </InputContainer>
  )
}

TextInput.defaultProps = {
  onChange: () => { },
  icon: faUser
}

TextInput.propTypes = {
  onChange: PropTypes.func,
  icon: PropTypes.object
}

export default TextInput