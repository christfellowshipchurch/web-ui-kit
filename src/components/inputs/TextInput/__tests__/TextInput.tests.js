import React from 'react'
import renderer from 'react-test-renderer'
import {
    configure, shallow, mount, render
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import TextInput from '../index'

configure({ adapter: new Adapter() })

describe('TextInput', () => {
    it('renders a Text Input without crashing', () => {
        const tree = renderer.create(
            <TextInput
                label="Label"
                description="Description" />
        )

        expect(tree).toMatchSnapshot()
    })

    it('renders a Text Input with a placeholder', () => {
        const tree = renderer.create(
            <TextInput
                label="Label"
                description="Description"
                placeholder="Placeholder" />
        )

        expect(tree).toMatchSnapshot()
    })

    it('renders a Text Input with a prefix', () => {
        const tree = renderer.create(
            <TextInput
                label="Label"
                description="Description"
                prefix={<div style={{ background: 'red', height: 20, width: 20 }}></div>} />
        )

        expect(tree).toMatchSnapshot()
    })

    it('renders a Text Input with a suffix', () => {
        const tree = renderer.create(
            <TextInput
                label="Label"
                description="Description"
                suffix={<div style={{ background: 'red', height: 20, width: 20 }}></div>} />
        )

        expect(tree).toMatchSnapshot()
    })

    it('renders a Text Input with an error', () => {
        const tree = renderer.create(
            <TextInput
                label="Label"
                description="Description"
                error="Here's an error" />
        )

        expect(tree).toMatchSnapshot()
    })

    it('renders a Text Input with using success', () => {
        const tree = renderer.create(
            <TextInput
                label="Label"
                description="Description"
                withSuccess />
        )

        expect(tree).toMatchSnapshot()
    })

    it('renders a disabled Text Input', () => {
        const tree = renderer.create(
            <TextInput
                label="Label"
                description="Description"
                disabled />
        )

        expect(tree).toMatchSnapshot()
    })

    it('simulates the click of a Text Input', () => {
        const wrapper = mount(
            <TextInput
                label="Label"
                description="Description" />
        )
        const input = wrapper.find('input')

        // Finds the input and clicks it
        input.simulate('focus')
        wrapper.update()

        expect(wrapper).toMatchSnapshot()

        // Removes focus
        input.simulate('blur')
        wrapper.update()

        expect(wrapper).toMatchSnapshot()
    })

    it('simulates the mouse enter and leave of a Text Input', () => {
        const wrapper = mount(
            <TextInput
                label="Label"
                description="Description" />
        )
        const input = wrapper.find('input')

        // Mouse Enters
        input.simulate('mouseenter')
        wrapper.update()

        expect(wrapper).toMatchSnapshot()

        // Mouse leaves
        input.simulate('mouseleave')
        wrapper.update()

        expect(wrapper).toMatchSnapshot()
    })

    it('simulates the mouse enter and leave of a Text Input', () => {
        const wrapper = mount(
            <TextInput
                label="Label"
                description="Description" />
        )
        const input = wrapper.find('input')

        // Types in the input
        input.simulate('change', { target: { value: 'My new value' } })
        wrapper.update()

        expect(wrapper).toMatchSnapshot()
    })
})