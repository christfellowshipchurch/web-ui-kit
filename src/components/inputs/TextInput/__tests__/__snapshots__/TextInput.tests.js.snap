// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders a Text Input with a placeholder 1`] = `
<div>
  <div
    className="d-flex flex-row align-items-end border-bottom border-light pb-2 px-1"
  >
    <div
      className="d-flex align-items-end justify-content-center"
      style={
        Object {
          "fontSize": 24,
          "minWidth": 30,
        }
      }
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-user fa-w-14 fa-1x "
        color="#525252"
        data-icon="user"
        data-prefix="fal"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M313.6 288c-28.7 0-42.5 16-89.6 16-47.1 0-60.8-16-89.6-16C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4zM416 464c0 8.8-7.2 16-16 16H48c-8.8 0-16-7.2-16-16v-41.6C32 365.9 77.9 320 134.4 320c19.6 0 39.1 16 89.6 16 50.4 0 70-16 89.6-16 56.5 0 102.4 45.9 102.4 102.4V464zM224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm0-224c52.9 0 96 43.1 96 96s-43.1 96-96 96-96-43.1-96-96 43.1-96 96-96z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="w-100 form-group mb-0 pt-3 px-2 p-relative"
    >
      <label
        className="m-0 input-label-sm"
      >
        Label
      </label>
      <input
        className="w-100 py-0"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        placeholder="Placeholder"
        type="text"
      />
    </div>
  </div>
  <small
    className="form-text text-muted"
  >
    Description
  </small>
</div>
`;

exports[`TextInput renders a Text Input with a prefix 1`] = `
<div>
  <div
    className="d-flex flex-row align-items-end border-bottom border-light pb-2 px-1"
  >
    <div
      className="d-flex align-items-end justify-content-center"
      style={
        Object {
          "fontSize": 24,
          "minWidth": 30,
        }
      }
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-user fa-w-14 fa-1x "
        color="#525252"
        data-icon="user"
        data-prefix="fal"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M313.6 288c-28.7 0-42.5 16-89.6 16-47.1 0-60.8-16-89.6-16C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4zM416 464c0 8.8-7.2 16-16 16H48c-8.8 0-16-7.2-16-16v-41.6C32 365.9 77.9 320 134.4 320c19.6 0 39.1 16 89.6 16 50.4 0 70-16 89.6-16 56.5 0 102.4 45.9 102.4 102.4V464zM224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm0-224c52.9 0 96 43.1 96 96s-43.1 96-96 96-96-43.1-96-96 43.1-96 96-96z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="w-100 form-group mb-0 pt-3 px-2 p-relative"
    >
      <label
        className="m-0 input-label"
      >
        Label
      </label>
      <input
        className="w-100 py-0"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        prefix={
          <div
            style={
              Object {
                "background": "red",
                "height": 20,
                "width": 20,
              }
            }
          />
        }
        type="text"
      />
    </div>
  </div>
  <small
    className="form-text text-muted"
  >
    Description
  </small>
</div>
`;

exports[`TextInput renders a Text Input with a suffix 1`] = `
<div>
  <div
    className="d-flex flex-row align-items-end border-bottom border-light pb-2 px-1"
  >
    <div
      className="d-flex align-items-end justify-content-center"
      style={
        Object {
          "fontSize": 24,
          "minWidth": 30,
        }
      }
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-user fa-w-14 fa-1x "
        color="#525252"
        data-icon="user"
        data-prefix="fal"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M313.6 288c-28.7 0-42.5 16-89.6 16-47.1 0-60.8-16-89.6-16C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4zM416 464c0 8.8-7.2 16-16 16H48c-8.8 0-16-7.2-16-16v-41.6C32 365.9 77.9 320 134.4 320c19.6 0 39.1 16 89.6 16 50.4 0 70-16 89.6-16 56.5 0 102.4 45.9 102.4 102.4V464zM224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm0-224c52.9 0 96 43.1 96 96s-43.1 96-96 96-96-43.1-96-96 43.1-96 96-96z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="w-100 form-group mb-0 pt-3 px-2 p-relative"
    >
      <label
        className="m-0 input-label"
      >
        Label
      </label>
      <input
        className="w-100 py-0"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        suffix={
          <div
            style={
              Object {
                "background": "red",
                "height": 20,
                "width": 20,
              }
            }
          />
        }
        type="text"
      />
    </div>
  </div>
  <small
    className="form-text text-muted"
  >
    Description
  </small>
</div>
`;

exports[`TextInput renders a Text Input with an error 1`] = `
<div>
  <div
    className="d-flex flex-row align-items-end border-bottom border-light pb-2 px-1"
  >
    <div
      className="d-flex align-items-end justify-content-center"
      style={
        Object {
          "fontSize": 24,
          "minWidth": 30,
        }
      }
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-user fa-w-14 fa-1x "
        color="#525252"
        data-icon="user"
        data-prefix="fal"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M313.6 288c-28.7 0-42.5 16-89.6 16-47.1 0-60.8-16-89.6-16C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4zM416 464c0 8.8-7.2 16-16 16H48c-8.8 0-16-7.2-16-16v-41.6C32 365.9 77.9 320 134.4 320c19.6 0 39.1 16 89.6 16 50.4 0 70-16 89.6-16 56.5 0 102.4 45.9 102.4 102.4V464zM224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm0-224c52.9 0 96 43.1 96 96s-43.1 96-96 96-96-43.1-96-96 43.1-96 96-96z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="w-100 form-group mb-0 pt-3 px-2 p-relative"
    >
      <label
        className="m-0 input-label"
      >
        Label
      </label>
      <input
        className="w-100 py-0"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        type="text"
      />
    </div>
    <div
      className="d-flex align-items-end justify-content-center"
      style={
        Object {
          "fontSize": 24,
          "minWidth": 30,
        }
      }
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-exclamation-circle fa-w-16 fa-1x "
        color="#cb045b"
        data-icon="exclamation-circle"
        data-prefix="fal"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 40c118.621 0 216 96.075 216 216 0 119.291-96.61 216-216 216-119.244 0-216-96.562-216-216 0-119.203 96.602-216 216-216m0-32C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm-11.49 120h22.979c6.823 0 12.274 5.682 11.99 12.5l-7 168c-.268 6.428-5.556 11.5-11.99 11.5h-8.979c-6.433 0-11.722-5.073-11.99-11.5l-7-168c-.283-6.818 5.167-12.5 11.99-12.5zM256 340c-15.464 0-28 12.536-28 28s12.536 28 28 28 28-12.536 28-28-12.536-28-28-28z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
  <small
    className="form-text text-muted"
  >
    Here's an error
  </small>
</div>
`;

exports[`TextInput renders a Text Input with using success 1`] = `
<div>
  <div
    className="d-flex flex-row align-items-end border-bottom border-light pb-2 px-1"
  >
    <div
      className="d-flex align-items-end justify-content-center"
      style={
        Object {
          "fontSize": 24,
          "minWidth": 30,
        }
      }
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-user fa-w-14 fa-1x "
        color="#525252"
        data-icon="user"
        data-prefix="fal"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M313.6 288c-28.7 0-42.5 16-89.6 16-47.1 0-60.8-16-89.6-16C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4zM416 464c0 8.8-7.2 16-16 16H48c-8.8 0-16-7.2-16-16v-41.6C32 365.9 77.9 320 134.4 320c19.6 0 39.1 16 89.6 16 50.4 0 70-16 89.6-16 56.5 0 102.4 45.9 102.4 102.4V464zM224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm0-224c52.9 0 96 43.1 96 96s-43.1 96-96 96-96-43.1-96-96 43.1-96 96-96z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="w-100 form-group mb-0 pt-3 px-2 p-relative"
    >
      <label
        className="m-0 input-label"
      >
        Label
      </label>
      <input
        className="w-100 py-0"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        type="text"
      />
    </div>
    <div
      className="d-flex align-items-end justify-content-center"
      style={
        Object {
          "fontSize": 24,
          "minWidth": 30,
        }
      }
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-check-circle fa-w-16 fa-1x "
        color="#1ec27f"
        data-icon="check-circle"
        data-prefix="fal"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm0 464c-118.664 0-216-96.055-216-216 0-118.663 96.055-216 216-216 118.664 0 216 96.055 216 216 0 118.663-96.055 216-216 216zm141.63-274.961L217.15 376.071c-4.705 4.667-12.303 4.637-16.97-.068l-85.878-86.572c-4.667-4.705-4.637-12.303.068-16.97l8.52-8.451c4.705-4.667 12.303-4.637 16.97.068l68.976 69.533 163.441-162.13c4.705-4.667 12.303-4.637 16.97.068l8.451 8.52c4.668 4.705 4.637 12.303-.068 16.97z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
  <small
    className="form-text text-muted"
  >
    Description
  </small>
</div>
`;

exports[`TextInput renders a Text Input without crashing 1`] = `
<div>
  <div
    className="d-flex flex-row align-items-end border-bottom border-light pb-2 px-1"
  >
    <div
      className="d-flex align-items-end justify-content-center"
      style={
        Object {
          "fontSize": 24,
          "minWidth": 30,
        }
      }
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-user fa-w-14 fa-1x "
        color="#525252"
        data-icon="user"
        data-prefix="fal"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M313.6 288c-28.7 0-42.5 16-89.6 16-47.1 0-60.8-16-89.6-16C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4zM416 464c0 8.8-7.2 16-16 16H48c-8.8 0-16-7.2-16-16v-41.6C32 365.9 77.9 320 134.4 320c19.6 0 39.1 16 89.6 16 50.4 0 70-16 89.6-16 56.5 0 102.4 45.9 102.4 102.4V464zM224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm0-224c52.9 0 96 43.1 96 96s-43.1 96-96 96-96-43.1-96-96 43.1-96 96-96z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="w-100 form-group mb-0 pt-3 px-2 p-relative"
    >
      <label
        className="m-0 input-label"
      >
        Label
      </label>
      <input
        className="w-100 py-0"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        type="text"
      />
    </div>
  </div>
  <small
    className="form-text text-muted"
  >
    Description
  </small>
</div>
`;

exports[`TextInput renders a disabled Text Input 1`] = `
<div>
  <div
    className="d-flex flex-row align-items-end border-bottom border-light pb-2 px-1"
  >
    <div
      className="d-flex align-items-end justify-content-center"
      style={
        Object {
          "fontSize": 24,
          "minWidth": 30,
        }
      }
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-user fa-w-14 fa-1x "
        color="#e6e6e6"
        data-icon="user"
        data-prefix="fal"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M313.6 288c-28.7 0-42.5 16-89.6 16-47.1 0-60.8-16-89.6-16C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4zM416 464c0 8.8-7.2 16-16 16H48c-8.8 0-16-7.2-16-16v-41.6C32 365.9 77.9 320 134.4 320c19.6 0 39.1 16 89.6 16 50.4 0 70-16 89.6-16 56.5 0 102.4 45.9 102.4 102.4V464zM224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm0-224c52.9 0 96 43.1 96 96s-43.1 96-96 96-96-43.1-96-96 43.1-96 96-96z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="w-100 form-group mb-0 pt-3 px-2 p-relative"
    >
      <label
        className="m-0 input-label"
      >
        Label
      </label>
      <input
        className="w-100 py-0"
        disabled={true}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        type="text"
      />
    </div>
    <div
      className="d-flex align-items-end justify-content-center"
      style={
        Object {
          "fontSize": 24,
          "minWidth": 30,
        }
      }
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-ban fa-w-16 fa-1x "
        color="#e6e6e6"
        data-icon="ban"
        data-prefix="fal"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zM103.265 408.735c-80.622-80.622-84.149-208.957-10.9-293.743l304.644 304.643c-84.804 73.264-213.138 69.706-293.744-10.9zm316.37-11.727L114.992 92.365c84.804-73.263 213.137-69.705 293.743 10.9 80.622 80.621 84.149 208.957 10.9 293.743z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
  <small
    className="form-text text-muted"
  >
    Description
  </small>
</div>
`;

exports[`TextInput simulates the click of a Text Input 1`] = `ReactWrapper {}`;

exports[`TextInput simulates the click of a Text Input 2`] = `ReactWrapper {}`;

exports[`TextInput simulates the mouse enter and leave of a Text Input 1`] = `ReactWrapper {}`;

exports[`TextInput simulates the mouse enter and leave of a Text Input 2`] = `ReactWrapper {}`;

exports[`TextInput simulates the mouse enter and leave of a Text Input 3`] = `ReactWrapper {}`;
